### PR TITLE
Add import command for csv files

### LINKED ISSUE #96: Add import file command

### PR DESCRIPTION
**Pull request by limjeremy496**
close #96

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                case "student":
+                    String studentID = values[2].trim();
+                    if (!studentID.isEmpty()) {
+                        Student student = new Student(name, new StudentID(studentID), phone, email, address, tagSet);
+                        if (!model.hasPerson(student)) {
+                            new AddStudentCommand(student).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Student ID for student category");
```
Comment (ruiming97) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:88]: Perhaps can add a test where the category is `student` and the `studentId` is missing to check if this exception is thrown

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                Name name = new Name(values[0].trim());
+                String category = values[1].trim();
+                Phone phone = new Phone(values[3].trim());
+                Email email = new Email(values[4].trim());
+                Address address = new Address(values[5].trim());
+                Set<Tag> tagSet = parseTags(values.length > 6 ? values[6].trim() : "");
+
+                switch (category.toLowerCase()) {
+                case "student":
+                    String studentID = values[2].trim();
```
Comment (ruiming97) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:80]: I think it should be `studentId` to follow the java coding standard

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                case "company":
+                    String industry = values[2].trim();
+                    if (!industry.isEmpty()) {
+                        Company company = new Company(name, new Industry(industry), phone, email, address, tagSet);
+                        if (!model.hasPerson(company)) {
+                            new AddCompanyCommand(company).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Industry for company category");
```
Comment (ruiming97) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:101]: Same here can add a test for this

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -89,6 +90,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case FilterTagCommand.COMMAND_WORD:
             return new FilterTagCommandParser().parse(arguments);
 
+        case ImportCommand.COMMAND_WORD:
```
Comment (ruiming97) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:93]: Can add a test for this too

**Review by ruiming97 — COMMENTED** [review]
Overall LGTM! Just need to add tests for some lines

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            + "Example: " + COMMAND_WORD + " /path/to/file.csv";
+    private final String filePath;
+
+    /**
+     * Creates an ImportCommand to import contacts from the specified CSV file path.
+     *
+     * @param filePath The path of the CSV file to import contacts from.
+     */
+    public ImportCommand(String filePath) {
+        requireNonNull(filePath);
```
Comment (JiaYi-Gallium369) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:44]: Good that you included defensive programming in your code!

File: src/main/java/seedu/address/logic/commands/ImportCommand.java (hunk 3 above)
Comment (JiaYi-Gallium369) [reply src/main/java/seedu/address/logic/commands/ImportCommand.java:101]: Yes, can maybe add an `assertThrows()` test

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    /**
+     * Returns the hash code for this ImportCommand.
+     *
+     * @return The hash code for this ImportCommand, based on the file path.
+     */
+    @Override
+    public int hashCode() {
+        return filePath.hashCode();
```
Comment (JiaYi-Gallium369) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:162]: Can maybe add a test case to check if the actual `hashCode` returned is equal with the expected one

**Review by JiaYi-Gallium369 — COMMENTED** [review]
LGTM overall

File: src/main/java/seedu/address/logic/commands/ImportCommand.java (hunk 3 above)
Comment (limjeremy496) [reply src/main/java/seedu/address/logic/commands/ImportCommand.java:101]: Thank you for the advice!

**Pull request merged**